### ISSUE TITLE
How to implement revoke certificate in Plugin

### ISSUE THREAD

**Issue by RcRonco** [Netflix/lemur#536]
* I did not see any way to ```revoke certificate``` in the lemur plugins interfaces. 
 I guess this is the job of the active/inactive button on the UI, but have not seen a way to implemented it.

* How i show to the user error from plugin, except from ```Internal Server error``` with raising Exception.

**Comment by kevgliss** [comment]
There is no current support for revocation in Lemur. We largely use Lemur as a self-service portal for developers and it was determined that revocation was a high risk activity we would not generally expose. Combined with the use of short-lived certificates and infrequency of revocation means that adding support was low value.

That being said I am open to contributions in this area. 

What kind of error are you considering showing the user, validation error or something else?

**Comment by RcRonco** [comment]
So the active certificate is just indicator not a button to revoke?
 
I would like to show validation errors to the users.

**Comment by kevgliss** [comment]
Yes, the active button is just a record keeping action and controls which certificates receive notifications. 

For validation errors you can always raise a custom exception `raise Exception('This is your error')` this will message will be propagated back to the user. 

For example:
https://github.com/Netflix/lemur/blob/master/lemur/plugins/lemur_verisign/plugin.py#L128

Dynamically raises the API error and gives a helpful message to the user. But this would be after the fact. 

If you are looking at providing feedback to a user in JS before request is submitted you can do something like:

https://github.com/Netflix/lemur/blob/master/lemur/plugins/lemur_aws/plugin.py#L52

Where the validation is a simple regex on the input value. They will then be shown the error as they are typing the value into the input.

**Comment by RcRonco** [comment]
I tried to raise Exception but it just showed ```Internal Server Error``` in the ui, for example:
https://github.com/RcRonco/lemur_vault/blob/master/lemur_vault/plugin.py#L50

The options dictionary is not an option for issuer plugin?

**Comment by kevgliss** [comment]
Ah, you are correct it will result in a `Internal Server Error` with  500 status code. It should however correctly show your message. Is that not the case?

**Comment by RcRonco** [comment]
It's not my case it just show red message with ```Internal Server error```.

I would like to show error on wrong parameters that plugins get, like TTL of certificate is too high.

**Comment by kevgliss** [comment]
Can you see if there is any more detail in the response? If you look inside chrome's inspector for instance, in the network details you should be able to see the whole response.

**Comment by RcRonco** [comment]
The result of this line:
https://github.com/RcRonco/lemur_vault/blob/master/lemur_vault/plugin.py#L50
Is the picture below, i can not expect my users to  check with the developer console what wrong with their input.
![image](https://cloud.githubusercontent.com/assets/13708679/20770762/7f1b261e-b74f-11e6-8e62-428442fdb042.png)

**Comment by kevgliss** [comment]
What is the the stack trace in the lemur.log? My guess is that you're hitting a different error, not the one expecting.

**Comment by RcRonco** [comment]
In the stack trace I see the expected exception message.

**Comment by kevgliss** [comment]
Can you post the stacktrace?

**Comment by RcRonco** [comment]
Sure I will post it tomorrow

**Comment by RcRonco** [comment]
```python
Traceback (most recent call last):
  File "/www/lemur/lemur/common/schema.py", line 136, in decorated_function
    resp = f(*args, **kwargs)
  File "/www/lemur/lemur/certificates/views.py", line 233, in post
    return service.create(**data)
  File "/www/lemur/lemur/certificates/service.py", line 214, in create
    cert_body, private_key, cert_chain = mint(**kwargs)
  File "/www/lemur/lemur/certificates/service.py", line 158, in mint
    cert_body, cert_chain = issuer.create_certificate(csr, kwargs)
  File "/www/lemur/lemur/plugins/lemur_vault/plugin.py", line 98, in create_certificate
    raise Exception('Vault Test')
Exception: Vault Test
Internal Error
Traceback (most recent call last):
  File "/www/lemur/lemur/common/schema.py", line 136, in decorated_function
    resp = f(*args, **kwargs)
  File "/www/lemur/lemur/certificates/views.py", line 233, in post
    return service.create(**data)
  File "/www/lemur/lemur/certificates/service.py", line 214, in create
    cert_body, private_key, cert_chain = mint(**kwargs)
  File "/www/lemur/lemur/certificates/service.py", line 158, in mint
    cert_body, cert_chain = issuer.create_certificate(csr, kwargs)
  File "/www/lemur/lemur/plugins/lemur_vault/plugin.py", line 98, in create_certificate
    raise Exception('Vault Test')
Exception: Vault Test

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/www/lemur/lib/python3.5/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/www/lemur/lib/python3.5/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/www/lemur/lib/python3.5/site-packages/flask_restful/__init__.py", line 462, in wrapper
    resp = resource(*args, **kwargs)
  File "/www/lemur/lib/python3.5/site-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/www/lemur/lib/python3.5/site-packages/flask_restful/__init__.py", line 572, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/www/lemur/lemur/auth/service.py", line 111, in decorated_function
    return f(*args, **kwargs)
  File "/www/lemur/lemur/common/schema.py", line 139, in decorated_function
    return dict(message=e.message), 500
AttributeError: 'Exception' object has no attribute 'message'
```

**Comment by kevgliss** [comment]
This was fixed in 0.4. Please make sure you have that release:
https://github.com/Netflix/lemur/commit/b74df2b3e407fcda10259f6d612d846ca1b9a47b
